pennylane/wires.py

```diff
@@ -493,6 +493,200 @@ def unique_wires(list_of_wires):
+
+        Args:
+            other (Wires): Another Wires object to perform the union with.
+
+        Returns:
+            Wires: A new Wires object representing the union of the two Wires objects.
+
+        Raises:
+            TypeError: If `other` is not an instance of Wires.
+
+        **Example**
+
+        >>> wires1 = Wires([1, 2, 3])
+        >>> wires2 = Wires([3, 4, 5])
+        >>> wires1.union(wires2)
+        Wires([1, 2, 3, 4, 5])
+        """
+        return Wires((set(self.labels) | set(_process(other))))
+
+    def __or__(self, other):
+        """Return the union of the current Wires object and either another Wires object or an
+        iterable that can be interpreted like a Wires object e.g. List.
+
+        Args:
+            other (Wires): Another Wires object to perform the union with.
+
+        Returns:
+            Wires: A new Wires object representing the union of the two Wires objects.
+
+        Raises:
+            TypeError: If `other` is not an instance of Wires.
+
+        **Example**
+
+        >>> wires1 = Wires([1, 2, 3])
+        >>> wires2 = Wires([3, 4, 5])
+        >>> wires1 | wires2
+        Wires([1, 2, 3, 4, 5])
+        """
+        return self.union(other)
+
+    def __ror__(self, other):
+        """Right-hand version of __or__."""
+        return self.union(other)
+
+    def intersection(self, other):
+        """Return the intersection of the current Wires object and either another Wires object or
+        an iterable that can be interpreted like a Wires object e.g. List.
+
+        Args:
+            other (Wires): Another Wires object to perform the intersection with.
+
+        Returns:
+            Wires: A new Wires object representing the intersection of the two Wires objects.
+
+        Raises:
+            TypeError: If `other` is not an instance of Wires.
+
+        **Example**
+
+        >>> wires1 = Wires([1, 2, 3])
+        >>> wires2 = Wires([2, 3, 4])
+        >>> wires1.intersection(wires2)
+        Wires([2, 3])
+        """
+
+        return Wires((set(self.labels) & set(_process(other))))
+
+    def __and__(self, other):
+        """Return the intersection of the current Wires object and either another Wires object or
+        an iterable that can be interpreted like a Wires object e.g. List.
+
+        Args:
+            other (Wires): Another Wires object to perform the intersection with.
+
+        Returns:
+            Wires: A new Wires object representing the intersection of the two Wires objects.
+
+        Raises:
+            TypeError: If `other` is not an instance of Wires.
+
+        **Example**
+
+        >>> wires1 = Wires([1, 2, 3])
+        >>> wires2 = Wires([2, 3, 4])
+        >>> wires1 & wires2
+        Wires([2, 3])
+        """
+        return self.intersection(other)
+
+    def __rand__(self, other):
+        """Right-hand version of __and__."""
+        return self.intersection(other)
```

Comment:
Should it be `other.intersection(self)`?

Reply:
It is a little confusing but this is correct. If you tried other.intersection(self) you would get an `NotImplementedError` (Or something similar).

Using the python docs as [reference](https://docs.python.org/3/reference/datamodel.html#object.__rand__), `__rand__` is only called if the left operand does not support `__and__` the operands are of different types. When `__rand__` is called, you should read from the right hand side to the left hand side. 

For example:
```
wires1 = Wires([0, 1]])

wires & [1, 2] # uses __and__, self = wires, other = [1, 2]
[1, 2] & wires # tries __and__ for [1, 2], then tries __rand__ for wires, therefore self = wires, other = [1, 2]
```

This particular implementation works for `Union` and `Intersection` since they are commutative. For non-commutative operations like `sub` we write a different function.